db/cassandra/src/main/java/org/commonjava/indy/cassandra/data/CassandraStoreDataManager.java

```diff
@@ -0,0 +1,348 @@
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@ApplicationScoped
+@Alternative
+public class CassandraStoreDataManager extends AbstractStoreDataManager
+{
+
+    private final Logger logger = LoggerFactory.getLogger( getClass() );
+
+    @Inject
+    CassandraStoreQuery storeQuery;
+
+    @Inject
+    IndyObjectMapper objectMapper;
+
+    @Override
+    protected StoreEventDispatcher getStoreEventDispatcher()
+    {
+        return null;
+    }
+
+    @Override
+    protected ArtifactStore getArtifactStoreInternal( StoreKey key )
+    {
+        DtxArtifactStore dtxArtifactStore = storeQuery.getArtifactStore( key.getPackageType(), key.getType(), key.getName() );
+
+        return toArtifactStore( dtxArtifactStore );
+    }
+
+    @Override
+    protected ArtifactStore removeArtifactStoreInternal( StoreKey key )
+    {
+        DtxArtifactStore dtxArtifactStore = storeQuery.removeArtifactStore( key.getPackageType(), key.getType(), key.getName() );
+        return toArtifactStore( dtxArtifactStore );
+    }
+
+    @Override
+    public void clear( ChangeSummary summary ) throws IndyDataException
+    {
+
+    }
+
+    @Override
+    public Set<ArtifactStore> getAllArtifactStores()
+    {
+        Set<DtxArtifactStore> dtxArtifactStoreSet = storeQuery.getAllArtifactStores();
+        Set<ArtifactStore> artifactStoreSet = new HashSet<>(  );
+        dtxArtifactStoreSet.forEach( dtxArtifactStore -> {
+            artifactStoreSet.add( toArtifactStore( dtxArtifactStore ) );
+        } );
+        return artifactStoreSet;
+    }
+
+    @Override
+    public Map<StoreKey, ArtifactStore> getArtifactStoresByKey()
+    {
+        Map<StoreKey, ArtifactStore> ret = new HashMap<>();
+        Set<ArtifactStore> artifactStoreSet = getAllArtifactStores();
+        artifactStoreSet.forEach( store -> {
+            ret.put( store.getKey(), store );
+        } );
+        return ret;
+    }
+
+    @Override
+    public boolean hasArtifactStore( StoreKey key )
+    {
+        ArtifactStore artifactStore = getArtifactStoreInternal( key );
+        return artifactStore != null;
+    }
+
+    @Override
+    public boolean isStarted()
+    {
+        return false;
+    }
+
+    @Override
+    public boolean isEmpty()
+    {
+        return false;
+    }
+
+    @Override
+    public Stream<StoreKey> streamArtifactStoreKeys()
+    {
+        return getAllArtifactStores().stream().map( ArtifactStore::getKey ).collect( Collectors.toSet() ).stream();
+    }
+
+    @Override
+    protected ArtifactStore putArtifactStoreInternal( StoreKey storeKey, ArtifactStore store )
+    {
+        DtxArtifactStore dtxArtifactStore = toDtxArtifactStore( storeKey, store );
+        storeQuery.createDtxArtifactStore( dtxArtifactStore );
+
+        return toArtifactStore( storeQuery.getArtifactStore( storeKey.getPackageType(), storeKey.getType(), storeKey.getName() ));
```

Comment:
Note that if we don't calculate any fields during the DTX update, we don't really need to re-query this (I don't think). We might just return the same object used to create the DtxArtifactStore, and save some performance.

Reply:
Yeah, there is no calculation.